### PR TITLE
MMDP-292 Add logout and missing login tests

### PR DESCRIPTION
**Pull request by bozicschucky**
**What does this PR do?**

Adds the missing logout component and refactors the login component to match the current team standards.

**Description of Task to be completed?**

- [x] Add the logout component.
- [x] Add the missing login tests.
- [x] Refactor the login code to match the new changes.

**How should this be manually tested?**
- Clone this branch `ch-improve-login-logout-MMDP-292`
- Run `yarn dev` to start both backend and frontend
- Navigate to ` localhost:1234/login` in your browser to test the frontend implementation
- The token will be stored in the local storage.
- In the browser go the ` localhost:1234/logout` route to log out a user.

**Any background context you want to provide?**

None

**What are the relevant Jira issues?**

[MMDP-292](https://viisaus.atlassian.net/browse/MMDP-292)
**Screenshots**
screenshot 2019-02-13 at 09 44 32

### CODE REVIEW

File: client/containers/Login/index.js [hunk 1]
```diff
@@ -38,18 +38,29 @@ export class Login extends Component {
...
+        {...this.props}
+      />
     );
   }
 }
 
 Login.propTypes = {
   loginUser: PropTypes.func.isRequired,
+  history: PropTypes.shape({}).isRequired,
+  LoginStatus: PropTypes.shape({}).isRequired,
```
Comment (Alameen688) [inline client/containers/Login/index.js]: Kindly change to camel case
```suggestion
  loginStatus: PropTypes.shape({}).isRequired,
```

File: client/containers/Login/index.js [hunk 2]
```diff
@@ -38,18 +38,29 @@ export class Login extends Component {
...
 };
 
 const mapStateToProps = (state) => ({
-  Login: state.response,
+  LoginStatus: state.loginReducer,
```
Comment (Alameen688) [inline client/containers/Login/index.js]: Kindly change to camel case
```suggestion
  loginStatus: state.loginReducer,
```

File: client/containers/Login/logout.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { LogOutUser } from '../../store/actions/auth/login';
+
+export class LogOut extends Component {
```
Comment (Alameen688) [inline client/containers/Login/logout.js]: Preferably, component should be named Logout
```suggestion
export class Logout extends Component {
```
https://www.computerhope.com/jargon/s/signoff.htm

File: client/containers/Login/logout.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  componentWillMount = () => {
+    // eslint-disable-next-line no-shadow
+    const { history, LogOutUser } = this.props;
+    LogOutUser();
```
Comment (Alameen688) [inline client/containers/Login/logout.js]: Kindly use Camel case or simply `logoutz
```suggestion
    logoutUser();
```

**Comment by bozicschucky** [comment]
@Alameen688 I am done with the requested changes.
More feedback will be appreciated.

**Pull request merged**